src/account/AccountStorage.sol

```diff
@@ -41,17 +41,19 @@ struct SelectorData {
     bool isPublic;
     // Whether or not a shared validation function may be used to validate this function.
     bool allowSharedValidation;
-    // How many times a `PRE_HOOK_ALWAYS_DENY` has been added for this function.
-    // Since that is the only type of hook that may overlap, we can use this to track the number of times it has
-    // been applied, and whether or not the deny should apply. The size `uint48` was chosen somewhat arbitrarily,
-    // but it packs alongside `plugin` while still leaving some other space in the slot for future packing.
-    uint48 denyExecutionCount;
-    // User operation validation and runtime validation share a function reference.
+    // The execution hooks for this function selector.
+    EnumerableSet.Bytes32Set executionHooks;
+    // Which validation functions are associated with this function selector.
     EnumerableSet.Bytes32Set validations;
```

Comment:
Thoughts on removing this entirely? What I'm thinking of:
1. Account tracks list of validations
2. Permissions framework (pre val hooks attached to validation fns) is responsible for what selectors a val fn can operate on

If I'm not wrong this should remove the need to do an existence check across the global validations set + the selector associated validation set

Reply:
There's some flexibility in the shape of #2 - a simple MA could just do:

```
struct ValidationData {
  bool isShared
  bool isSignatureValidation
  mapping(bytes4 selector -> bool canValidate) canValidateSelector
  EnumerableSet.Bytes32Set preValidationHooks
}
```

to support 2 modes of validation functions - those that can be used either globally, or just for a single selector

Reply:
Without selector-associated validations, every validation would need a pre-hook just to do selector associations - it's one of those things that benefits from being builtin, but yes I agree we should be careful in what we do this for.

I actually thought about making the refactor to the format of `mapping(bytes4 selector -> bool canValidate) canValidateSelector`, but there's one reason we can't do it just yet - the current interface of our loupe functions.

Right now, we have `function getValidations(bytes4 selector) external view returns (FunctionReference[] memory)`, which returns the validations per selector. Even if we made the `canValidateSelector` enumerable, we wouldn't be able to implement this interface efficiently, we'd need to search every single validation function.

But, if we change the view function to `function getSelectors(FunctionReference validation) external view returns (bytes4[] memory)`, then we could make this refactor.

We should revisit this in the future, and decide which of these functions it actually makes sense to implement.

Reply:
We should capture this conversation somewhere. It is very meaningful, but is not in this PR's scope. @howydev Seems you proposed it. Maybe even capture as a proposal to the issues list.

In this world where there are only global validations, how are native functions' validation handled?
Would those native functions just parts of `canValidateSelector` on some of the validations? (Note, right now, default/global validators can validate any native functions).

Reply:
Looking at the code below, seems we don't really support `default/global validators can validate any native functions` natively in code. Users still need to setup by adding the validation functions to each native functions. Did I miss anything?

If the above is the case, then there is no reason to have two set of pools (default, and selector specific), we should combine them.

Reply:
I discussed this with @fangting-alchemy in a call, documenting here for others:

>Users still need to setup by adding the validation functions to each native functions.

This is not necessary, once a validation function is marked as `isDefault`, it will apply to any function allowed by default validation in [`_defaultValidationAllowed`](https://github.com/erc6900/reference-implementation/blob/4d2390be98f38fd71e1b28e7b0a52ea141088fe9/src/account/UpgradeableModularAccount.sol#L624), which includes the native functions.

The ability to pass in selectors to `installValidation` is how to install the validation as a non-default validation function. This allows users to install a validation as both a default validation and as a selector-associated one for multiple selectors at once, but in practice I expect most users to do one or the other (either default, or associated with some selectors).

Reply:
Related thought - have we considered making `allowDefaultValidation` configurable for native functions too? How would we reliably prevent validating into `uninstallPlugin` if we want to enforce a time delay? I guess we can just use a pre exec hook instead of a pre validation hook.

Is there ever a case where users might want to opt out of default validation for some subset of native functions? Hmm...